core/src/main/java/com/google/errorprone/refaster/UEnhancedForLoop.java

```diff
@@ -39,7 +37,69 @@
 abstract class UEnhancedForLoop extends USimpleStatement implements EnhancedForLoopTree {
   public static UEnhancedForLoop create(
       UVariableDecl variable, UExpression elements, UStatement statement) {
-    return new AutoValue_UEnhancedForLoop(variable, elements, (USimpleStatement) statement);
+    // On JDK 20 and above the `EnhancedForLoopTree` interface contains a additional method
+    // `getDeclarationKind()`, referencing a  type not available prior to JDK 20. AutoValue
+    // generates a corresponding field and accessor for this property. Here we find and invoke the
+    // generated constructor with the appropriate arguments, depending on context.
+    // See https://github.com/openjdk/jdk20/commit/2cb64a75578ccc15a1dfc8c2843aa11d05ca8aa7.
+    // TODO: Simplify this logic once JDK 19 and older are no longer supported.
+    return isCompiledWithJdk20Plus()
+        ? createJdk20PlusEnhancedForLoop(variable, elements, statement)
+        : createPreJdk20EnhancedForLoop(variable, elements, statement);
```

Comment:
Not happy with all the reflection going on here, but found no better way.

Reply:
We've mostly been using reflection for similar workarounds recently.

I've been thinking about whether could use Multi-Release jars. At some point it may be worth branches source files like this when there are significant API changes, to avoid the reflection. But at least the last time I looked into it, setting maven up to build mr-jars didn't seem trivial.

Reply:
Using Multi-Release JARs might be a nice idea indeed! I currently don't have any experience with those, so can't comment on the effort required for Maven. (But this does sound like a nice topic to dive into; might also be nice for Error Prone Support, where we're also discussing how to depend on/use newer JDKs while still supporting JDK 11.)